src/dataflow/src/render/sinks.rs

```diff
@@ -205,13 +207,30 @@ where
 
         match sink.connector.clone() {
             SinkConnector::Kafka(c) => {
+                // Extract handles to the relevant source timestamp histories the sink
+                // needs to hear from before it can write data out to Kafka.
+                let mut source_ts_histories = Vec::new();
+
+                if let Some(source_ids) = &c.exactly_once {
+                    for id in source_ids.iter() {
+                        if let Some(history) = render_state.ts_histories.get(id) {
+                            let mut history_bindings = history.clone();
+                            // We don't want these to block compaction
```

Comment:
this comment could also be a lot more descriptive -- basically these handles exist for peer source instances to indicate "ive read up to here, pls don't compact past that" but in this case the sinks only care about the durability frontier. maybe theres a better idiom to only share that